lib/helper/REST.js

```diff
@@ -41,6 +41,54 @@ class REST extends Helper {
     }
   }
 
+  /**
+   * Executes unirest request
+   *
+   * @param {*} request
+   */
+  async _executeRequest(request) {
+    if (headers && headers.auth) {
+      request.auth(headers.auth);
+    }
+    request.timeout(this.options.timeout);
+    request.followRedirect(this.options.followRedirect);
+    const response = await executeRequest(request);
+    this._cleanRequestHeaders();
+    return response;
+  }
+
+  /**
+   * Changes headers to default if reset headers option is true
+   *
+   */
+  _cleanRequestHeaders() {
+    if (this.options.resetHeaders) {
+      headers = Object.assign({}, this.options.defaultHeaders);
+    }
+  }
+
+  /**
+   * Generates url based on format sent (takes endpoint + url if latter lacks 'http')
+   *
+   * @param {*} url
+   */
+  _url(url) {
+    return url.indexOf('http') === -1 ? this.options.endpoint + url : url;
+  }
+
+  /**
+   * Set whether 3XX request should automatically redirect
+   *
+   * ```js
+   * I.setRequestRedirects(); // To enable auto-redirects
+   * ```
+   *
+   * @param {boolean} autoRedirect
+   */
+  setRequestRedirects(autoRedirect = true) {
```

Comment:
I'm against using booleans as parameters in API as they are not so user-friendly. The `autoRedirect` won't be visible in console. Better to make functions with different names:

```
I.amFollowingRequestRedirects();
I.amNotFollowingRequestRedirects();
```

Reply:
Agreed! 👍